java/vector/src/main/java/org/apache/arrow/vector/BaseIntVector.java

```diff
@@ -23,7 +23,13 @@
 public interface BaseIntVector extends ValueVector {
 
   /**
-   * set value at specific index, note this value may need to be need truncated.
+   * Sets the value at index, note this value may need to be need truncated.
    */
   void setWithPossibleTruncate(int index, long value);
```

Comment:
Do we also need a setSafe method here?

Reply:
Thanks, this method is implemented with setSafe in sub classes :)

Reply:
I see. Thanks. I am not sure if we should use separate APIs for set and setSafe methods.

Reply:
Lets update the javadoc to indicate this is the safe version.  Unfortuantely, I think this made it into the 0.14.0, so we should try not to rename it.  We can always add an unsafe version later.

Reply:
Sure, thanks, will also add a method like "setUnsafeWithPossibleTruncate"

Reply:
Thanks for your efforts, PR updated.